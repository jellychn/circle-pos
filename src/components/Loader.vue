<template>
  <div class="loader">
    <span class="spinner" />
  </div>
</template>

<style lang="scss" scoped>
.loader {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);

  .spinner {
    width: 60px;
    height: 60px;
    border: 5px solid var(--vt-c-white);
    border-bottom-color: var(--vt-c-red);
    border-radius: 50%;
    display: inline-block;
    box-sizing: border-box;
    animation: rotation 1s linear infinite;
  }
}

@keyframes rotation {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
</style>