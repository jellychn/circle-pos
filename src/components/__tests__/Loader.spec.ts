import { shallowMount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import Loader from '@/components/Loader.vue'

describe('Loader.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Loader)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
