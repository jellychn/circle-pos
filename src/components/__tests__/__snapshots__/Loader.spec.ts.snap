// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader.vue > matches snapshot 1`] = `"<div data-v-8ec46d9a="" class="loader"><span data-v-8ec46d9a="" class="spinner"></span></div>"`;
